sentry_sdk/integrations/boto3.py

```diff
@@ -0,0 +1,110 @@
+from __future__ import absolute_import
+
+from sentry_sdk import Hub
+from sentry_sdk.integrations import Integration
+from sentry_sdk.tracing import Span
+
+from sentry_sdk._functools import partial
+from sentry_sdk._types import MYPY
+
+if MYPY:
+    from typing import Any
+    from typing import Dict
+    from typing import Optional
+    from typing import Type
+
+
+class Boto3Integration(Integration):
+    identifier = "boto3"
+
+    @staticmethod
+    def setup_once():
+        # type: () -> None
+        from botocore.client import BaseClient  # type: ignore
+        from botocore.response import StreamingBody  # type: ignore
+
+        if MYPY:
+            from botocore.awsrequest import AWSRequest  # type: ignore
+
+        def sentry_request_created(service_id, request, operation_name, **kwargs):
+            # type: (str, AWSRequest, str, **Any) -> None
+            hub = Hub.current
+            description = "aws.%s.%s" % (service_id, operation_name)
+            span = hub.start_span(
+                hub=hub,
+                op="aws.request",
+                description=description,
+            )
+            span.set_tag("aws.service_id", service_id)
+            span.set_tag("aws.operation_name", operation_name)
+            span.set_data("aws.request.url", request.url)
+
+            # We do it in order for subsequent http calls/retries be
+            # attached to this span.
+            span.__enter__()
+
+            # request.context is an open-ended data-structure
+            # where we can add anything useful in request life cycle.
+            request.context["_sentrysdk_span"] = span
+
+        def sentry_after_call(context, parsed, **kwargs):
+            # type: (Dict[str, Any], Dict[str, Any], **Any) -> None
+            span = context.get("_sentrysdk_span")  # type: Optional[Span]
+            if span is None:
+                return
+            span.__exit__(None, None, None)
+
+            body = parsed.get("Body")
+            if not isinstance(body, StreamingBody):
+                return
+
+            streaming_span = span.start_child(
+                op="aws.request.stream",
+                description=span.description,
+            )
+
+            orig_read = body.read
+            orig_close = body.close
+
+            def sentry_streaming_body_read(amt=None):
```

Comment:
unless you need to interpret the arguments please use `*args, **kwargs` (in both read and close) to protect against api changes